src/presence_analyzer/utils.py

```diff
@@ -35,6 +40,30 @@ def inner(*args, **kwargs):
     return inner
 
 
+def cache(ttl=600):
+    """
+    Stores result of get_data() function in cache.
+    :param ttl: time to live of cached result in seconds.
+    """
+    def _cache(_function):
+        def ttl_is_over(creation_timestamp):
+            return creation_timestamp + ttl < time.time()
+
+        def __cache():
+            with _LOCK:
+                if _storage and not ttl_is_over(_storage['creation_timestamp']):
+                    result = _storage['result']
+                else:
+                    result = _function()
+                    _storage['result'] = result
```

Comment:
It should be reusable, u should store sth like: 
`{func: {data: ..., creation/expiration/etc: ...}`